components/infobox/wikis/rainbowsix/infobox_weapon_custom.lua

```diff
@@ -6,65 +6,55 @@
 -- Please see https://github.com/Liquipedia/Lua-Modules to contribute
 --
 
-local Weapon = require('Module:Infobox/Weapon')
+local Array = require('Module:Array')
 local Class = require('Module:Class')
-local Builder = require('Module:Infobox/Widget/Builder')
-local Injector = require('Module:Infobox/Widget/Injector')
 local Cell = require('Module:Infobox/Widget/Cell')
-local Array = require('Module:Array')
+local Lua = require('Module:Lua')
 local OperatorIcon = require('Module:OperatorIcon')
 local Table = require('Module:Table')
 
-local CustomWeapon = Class.new()
+local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+local Weapon = require('Module:Infobox/Weapon', {requireDevIfEnabled = true})
+
+local CustomWeapon = Class.new(Weapon)
 local CustomInjector = Class.new(Injector)
 
 local _SIZE_OPERATOR = '25x25px'
 
-local _weapon
-local _args
-
 ---@param frame Frame
 ---@return Html
 function CustomWeapon.run(frame)
-	local weapon = Weapon(frame)
-	_weapon = weapon
-	_args = _weapon.args
+	local weapon = CustomWeapon(frame)
 
-	weapon.addToLpdb = CustomWeapon.addToLpdb
-	weapon.createWidgetInjector = CustomWeapon.createWidgetInjector
+	weapon:setWidgetInjector(CustomInjector(patch))
 	return weapon:createInfobox()
 end
 
----@return WidgetInjector
-function CustomWeapon:createWidgetInjector()
-	return CustomInjector()
-end
-
 ---@param widgets Widget[]
 ---@return Widget[]
-function CustomInjector:addCustomCells(widgets)
-	-- Operators
-	table.insert(widgets,
-		Builder{
-			builder = function()
-				local operatorIcons = Array.map(Weapon:getAllArgsForBase(_args, 'operator'),
-					function(operator, _)
-						return OperatorIcon.getImage{operator, size = _SIZE_OPERATOR}
-					end
-				)
-				return {
-					Cell{
-						name = #operatorIcons > 1 and 'Operators' or 'Operator',
-						content = {
-							table.concat(operatorIcons, '&nbsp;')
-						}
-					}
-				}
-			end
-		})
+function CustomInjector:parse(id, widgets)
+	if id == 'custom' then
+		return {
+			Cell{name = 'Operators', content = self:_getOperators()},
+		}
+	end
 	return widgets
 end
 
+---@return string[]
+function CustomWeapon:_getOperators()
+	local foundArgs = self:getAllArgsForBase(self.args, 'operator')
+
+	local operators = {}
+	for _, item in ipairs(foundArgs) do
+		local operator = Template.safeExpand(mw.getCurrentFrame(), 'Operator/' .. item, nil, '')
+		if not String.isEmpty(operator) then
+			table.insert(operators, operator)
+		end
+	end
+	return operators
```

Comment:
NO!!!
why go from module to shitty template expand???
also code style wise not good
```suggestion
	local operatorIcons = Array.map(self:getAllArgsForBase(self.args, 'operator'), function(operator, _)
		return OperatorIcon.getImage{operator, size = _SIZE_OPERATOR}
	end)

	return table.concat(operatorIcons, '&nbsp;')
```